.ci/CI/src/Bootloader.jl

```diff
@@ -81,18 +74,74 @@ function main()
-        append_custom_dependency_urls_from_git_message!(custom_dependency_urls)
-        append_custom_dependency_urls_from_env_var!(custom_dependency_urls)
+        if target_branch != "main"
+            append_custom_dependency_urls_from_git_message!(custom_dependency_urls)
+            append_custom_dependency_urls_from_env_var!(custom_dependency_urls)
+        end
 
-        add_integration_test_job_yaml!(
-            get(job_yamls, "output-cpu", job_yamls["stdout"]),
-            test_package,
-            target_branch,
-            custom_dependency_urls.integ,
-            tools_git_repo,
+        integration_test_package_names = get_qed_integration_test_package_names(
+            test_package, custom_dependency_urls.integ
         )
+
+        for (julia_version_type_name, platform) in tests_configurations[IntegrationTest]
+            output_yaml = get_output_job_yaml(job_yamls, platform)
+
+            for integration_package_name in integration_test_package_names
+                integration_test_repo = GitRepoAddress(
+                    get(
+                        custom_dependency_urls.integ,
+                        integration_package_name,
+                        "https://github.com/QEDjl-project/$(integration_package_name).jl.git"
+                    )
+                )
+
+                add_integration_test_job_yaml!(
+                    output_yaml,
+                    test_package,
+                    true, # setup dev env
+                    false, # can fail
+                    integration_package_name,
+                    integration_test_repo,
+                    julia_version_type_name,
+                    platform,
+                    tools_git_repo
+                )
+
+                # Handles the case of merging in the main branch. If we want to merge in the main branch,
+                # we do it because we want to publish the package. Therefore, we need to be sure that there
+                # is an existing version of the dependent QED packages that works with the new version of
+                # the package we want to release. The integration tests are tested against the development
+                # branch and the release version.
+                #  - The dev branch version must pass, as this means that the latest version of the other
+                #    QED packages is compatible with our release version.
+                #  - The release version integration tests may or may not pass.
+                #    1. If all of these pass, we will not need to increase the minor version of this package.
+                #    2. If they do not all pass, the minor version must be increased and the failing packages
+                #    must also be released later with an updated compat entry.
+                #    In either case the release can proceed, as the released packages will continue to work
+                #    because of their current compat entries.
```

Comment:
This is good documentation, could this be put in the actual docs too? I'm not sure if it's in there yet.